packages/scheduler/src/forks/SchedulerHostConfig.post-task.js

```diff
@@ -5,246 +5,117 @@
-    return Date.now() - initialTime;
-  };
-  requestHostCallback = function(cb) {
-    if (_callback !== null) {
-      // Protect against re-entrancy.
-      setTimeout(requestHostCallback, 0, cb);
-    } else {
-      _callback = cb;
-      setTimeout(_flushCallback, 0);
-    }
-  };
-  cancelHostCallback = function() {
-    _callback = null;
-  };
-  requestHostTimeout = function(cb, ms) {
-    _timeoutID = setTimeout(cb, ms);
-  };
-  cancelHostTimeout = function() {
-    clearTimeout(_timeoutID);
-  };
-  shouldYieldToHost = function() {
-    return false;
-  };
-  requestPaint = forceFrameRate = function() {};
+let getNow;
+if (typeof perf === 'object' && typeof perf.now === 'function') {
+  getNow = () => perf.now();
 } else {
-  // Capture local references to native APIs, in case a polyfill overrides them.
-  const performance = window.performance;
-  const Date = window.Date;
-  const setTimeout = window.setTimeout;
-  const clearTimeout = window.clearTimeout;
+  const initialTime = date.now();
+  getNow = () => date.now() - initialTime;
+}
 
-  if (typeof console !== 'undefined') {
-    // TODO: Scheduler no longer requires these methods to be polyfilled. But
-    // maybe we want to continue warning if they don't exist, to preserve the
-    // option to rely on it in the future?
-    const requestAnimationFrame = window.requestAnimationFrame;
-    const cancelAnimationFrame = window.cancelAnimationFrame;
-    // TODO: Remove fb.me link
-    if (typeof requestAnimationFrame !== 'function') {
-      // Using console['error'] to evade Babel and ESLint
-      console['error'](
-        "This browser doesn't support requestAnimationFrame. " +
-          'Make sure that you load a ' +
-          'polyfill in older browsers. https://fb.me/react-polyfills',
-      );
-    }
-    if (typeof cancelAnimationFrame !== 'function') {
-      // Using console['error'] to evade Babel and ESLint
-      console['error'](
-        "This browser doesn't support cancelAnimationFrame. " +
-          'Make sure that you load a ' +
-          'polyfill in older browsers. https://fb.me/react-polyfills',
-      );
-    }
+let isMessageLoopRunning = false;
+let scheduledHostCallback = null;
+let taskTimeoutID = -1;
+
+// Scheduler periodically yields in case there is other work on the main
+// thread, like user events. By default, it yields multiple times per frame.
+// It does not attempt to align with frame boundaries, since most tasks don't
+// need to be frame aligned; for those that do, use requestAnimationFrame.
+let yieldInterval = 5;
+let deadline = 0;
+
+// `isInputPending` is not available. Since we have no way of knowing if
+// there's pending input, always yield at the end of the frame.
+export const shouldYieldToHost = function() {
+  return getNow() >= deadline;
+};
+
+// Since we yield every frame regardless, `requestPaint` has no effect.
+export const requestPaint = function() {};
+
+export const forceFrameRate = function(fps) {
```

Comment:
This was added a few years ago for lower end devices but I'm not sure we even need it anymore.

Let's just leave this as a noop for now.